src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector.cs

```diff
@@ -61,6 +61,7 @@ public override BoundStatement InstrumentExpressionStatement(BoundExpressionStat
                     case SyntaxKind.BaseConstructorInitializer:
                     case SyntaxKind.ThisConstructorInitializer:
                         var init = (ConstructorInitializerSyntax)original.Syntax;
+                        Debug.Assert(init.Parent is object);
```

Comment:
Confused why this would be necessary but the line above was able to cast without adding a `?` to the type. If `init` was indeed maybe `null` here I would expect that the type cast above would have to say that as well.

Reply:
The problem is not that `init` is maybe-null, but that `init.Parent` is.

---
In reply to: [414135650](https://github.com/dotnet/roslyn/pull/43025#discussion_r414135650) [](ancestors = 414135650)